lib/gtfs/stop_time.ex

```diff
@@ -4,14 +4,12 @@ defmodule Gtfs.StopTime do
 
   A timepoint is a key stop along a route. In GTFS, timepoints are known as "checkpoints".
   """
-  alias Gtfs.Csv
-  alias Gtfs.Stop
-  alias Gtfs.Trip
+  alias Gtfs.{Csv, Data, Stop, Timepoint, Trip}
 
   @type t :: %__MODULE__{
           stop_id: Stop.id(),
           time: Util.Time.time_of_day(),
-          timepoint_id: timepoint_id() | nil
+          timepoint: Timepoint.t() | nil
```

Comment:
I think `stop_time` should still have a `Timepoint.id()`. Then in the rare case we need the full timepoint (I think just for getting the list of full timepoints for a route), we look up the ids in a separate table that forms a single source of truth. For most things we do with timepoint ids (e.g. measuring a bus's progress), having an id reference would be slimmer (important since all the stop_times really bog us down) and easier to work with when doing equality checks.

Reply:
What do you mean by "slimmer" and "bog us down"? I'm not sure I understand the problem you are trying to solve, but adding a store for these and an additional lookup for every vehicle on every update seems pretty expensive when this data is only used as a part of stop times.

Reply:
The lookup wouldn't happen on every update. The vehicles don't need to know the timepoint names. We only need it for routes. So, once on startup, we would get the `%{Route.id() => [Timepoint.id()]}` map as we do on master, then look those up into a `%{Route.id() => [Timepoint.t()]}` map.

I guess, after we store the timepoints_by_route, we might not even need to save the full timepoints with names anywhere else.